### ISSUE TITLE
Password fields should not deny spaces in password

### ISSUE THREAD

**Issue by gnunicorn** [acterglobal/a3#1809]
As noted  in https://github.com/acterglobal/a3/pull/1676#discussion_r1625670798_ we are currently denying spaces ` ` in our password fields, but the spec doesn't actually say that we should. Which means you could have added spaces in your password with other clients and we'd outright deny you entering your proper password.

However, the reason we introduce:
```
          inputFormatters: [
            FilteringTextInputFormatter.deny(RegExp(r'\s')),
          ],
```

was because people accidentally added spaces to their password, didn't realize it and then complained, they couldn't log in. We have improved the situation quite a bit since with allowing users to toggle the visibility of the password text to proof-read. However, spaces are still hidden to the user.

Instead we should warn the user if they are about to submit a password that is potentially problematic:
 - show a leading capslock icon when they have capslock on while the password field has focus
  - show a warning message below the text field that they entered spaces either at the beginning or end with a quick-action (inline-text button) to remove them.

**Comment by kumarpalsinh25** [comment]
@gnunicorn 
Is this issue is still persists? Can you check and confirm?

**Comment by gnunicorn** [comment]
@kumarpalsinh25 no. the password input field still denies `\s` for some reason.
